src/sentry/web/frontend/auth_login.py

```diff
@@ -95,39 +243,238 @@ def get_register_form(self, request: Request, initial=None):
+        self, request: Request, register_form: RegistrationForm, organization: Organization
+    ) -> HttpResponseRedirect:
+        """Given a valid register form, redirects the user to their next step."""
 
-    def get_join_request_link(self, organization):
-        if not organization:
-            return None
+        user = register_form.save()
+        user.send_confirm_emails(is_new_user=True)
+        user_signup.send_robust(sender=self, user=user, source="register-form", referrer="in-app")
 
-        if organization.get_option("sentry:join_requests") is False:
-            return None
+        # HACK: grab whatever the first backend is and assume it works
+        user.backend = settings.AUTHENTICATION_BACKENDS[0]
+        self._handle_login(request=request, user=user, organization=organization)
 
-        join_request_link_viewed.send_robust(sender=self, organization=organization)
+        # can_register should only allow a single registration
+        request.session.pop("can_register", None)
+        request.session.pop("invite_email", None)
 
-        return reverse("sentry-join-request", args=[organization.slug])
+        # Attempt to directly accept any pending invites
+        invite_helper = ApiInviteHelper.from_session(
+            request=request,
+        )
+
+        # In single org mode, associate the user to the only organization.
+        #
+        # XXX: Only do this if there isn't a pending invitation. The user
+        # may need to configure 2FA in which case, we don't want to make
+        # the association for them.
+        if settings.SENTRY_SINGLE_ORGANIZATION and not invite_helper:
+            self.add_user_to_organization(user=user)
+
+        if invite_helper and invite_helper.valid_request:
+            return self.accept_invite_and_redirect_to_org(
+                request=request, invite_helper=invite_helper
+            )
+
+        return self.redirect(url=self.get_post_register_url(request=request))
+
+    def add_user_to_organization(self, user: User) -> None:
+        """Adds a user to their default organization as a member."""
+        organization = Organization.get_default()
+        organization_service.add_organization_member(
+            organization_id=organization.id,
+            default_org_role=organization.default_role,
+            user_id=user.id,
+        )
+
+    def accept_invite_and_redirect_to_org(
+        self, request: Request, invite_helper: ApiInviteHelper
+    ) -> HttpResponseRedirect:
+        """Accepts an invite on behalf of a user and redirects them to their org login"""
+        invite_helper.accept_invite()
+        org_slug = invite_helper.invite_context.organization.slug
+        self.determine_active_organization(request=request, organization_slug=org_slug)
+        response = self.redirect_to_org(request=request)
+        remove_invite_details_from_session(request=request)
+        return response
+
+    def get_post_register_url(self, request: Request) -> str:
```

Comment:
oof 'post' here meaning "after" not `POST` 🙃 

can we think of a better word here?
'get_success_register_url'? eh...